azure-container-apps-mtls-certificate-filter/azure-container-apps-mtls-certificate-filter-jakarta/src/main/java/com/microsoft/sample/security/validator/X509CertificateValidator.java

```diff
@@ -0,0 +1,64 @@
+package com.microsoft.sample.security.validator;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+import java.io.FileInputStream;
+import java.security.KeyStore;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+import java.util.List;
+
+@Service
+public class X509CertificateValidator {
+
+    private X509TrustManager trustManager;
+
+    public X509CertificateValidator(@Value("${trust-store.path}") String trustStorePath,
+                                    @Value("${trust-store.password}") String trustStorePassword) throws Exception {
+        // Load the trust store
+        KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+        try (FileInputStream trustStoreStream = new FileInputStream(trustStorePath)) {
+            trustStore.load(trustStoreStream, trustStorePassword.toCharArray());
+        }
+
+        // Initialize the TrustManagerFactory with the trust store
+        TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        trustManagerFactory.init(trustStore);
+
+        // Get the X509TrustManager from the factory
+        for (javax.net.ssl.TrustManager tm : trustManagerFactory.getTrustManagers()) {
+            if (tm instanceof X509TrustManager) {
+                this.trustManager = (X509TrustManager) tm;
+                break;
+            }
+        }
+
+        if (this.trustManager == null) {
+            throw new IllegalStateException("No X509TrustManager found");
+        }
+    }
+
+    public void validate(X509Certificate[] certificateChain) throws CertificateException {
```

Comment:
According to the top level README, the validator is only a basic reference implementation. Let's add some comments in the source file to clarify what we are doing, and state that the customer needs to modify the logic according to their need.

Same to the other copies.